include/ocpn_plugin.h

```diff
@@ -168,6 +168,7 @@ class PlugIn_Position_Fix_Ex {
   double Var;  // Variation, typically from RMC message
   double Hdm;
   double Hdt;
+  // The time obtained from the most recent GNSS message, or the system time if the GNSS watchdog has expired.
   time_t FixTime;
```

Comment:
| Plugin | Use `PlugIn_Position_Fix*.FixTime`? |
|--------|--------|
| grib_pi | yes |
| dashboard_pi | yes |
| weather_routing_pi | no |

Reply:
I used a helper script to clone all plugins and analyze the `FixTime` dependencies.

```python
#!/usr/bin/env python3

import xml.etree.ElementTree as ET
import subprocess
import os
import argparse
import re

def get_clone_url(homepage_url):
    """
    Convert a GitHub homepage URL to a Git clone URL.
    
    :param homepage_url: GitHub repository homepage URL
    :return: Git clone URL (HTTPS)
    """
    homepage_url = homepage_url.rstrip('/')
    
    if not homepage_url.startswith('https://github.com/'):
        raise ValueError(f"Unsupported repository URL: {homepage_url}")
    
    match = re.match(r'https://github.com/([^/]+)/([^/]+)', homepage_url)
    if not match:
        raise ValueError(f"Invalid GitHub URL format: {homepage_url}")
    
    username, repo_name = match.groups()
    
    return f"https://github.com/{username}/{repo_name}.git"

def process(name, source):
    """
    Clone the plugin repository into the ./all-plugins directory if not already cloned.
    
    :param name: Name of the plugin
    :param source: GitHub repository homepage URL
    :return: Tuple of (name, git_url) if successful, None otherwise
    """
    clone_path = os.path.join("all-plugins", name)
    if os.path.exists(clone_path):
        print(f"Skipping {name}: Already cloned")
        return name, get_clone_url(source)

    print(f"Processing plugin: {name}")
    try:
        clone_url = get_clone_url(source)
        subprocess.run(["git", "clone", clone_url, clone_path], check=True)
        print(f"Successfully cloned {name} from {clone_url} into {clone_path}")
        return name, clone_url
    except ValueError as e:
        print(f"Error processing {name}: {str(e)}")
    except subprocess.CalledProcessError as e:
        print(f"Failed to clone {name} from {source}. Error: {e}")
    
    return None

def generate_markdown_table(plugins_info):
    """
    Generate a markdown table from the plugins information.
    
    :param plugins_info: List of tuples containing (name, git_url) for each plugin
    :return: String containing the markdown table
    """
    # Sort the plugins_info list by plugin name (case-insensitive)
    sorted_plugins = sorted(plugins_info, key=lambda x: x[0].lower())

    table = "| Plugin Name | Use FixTime | time_t == 0 special case | Git URL |\n"
    table += "|-------------|-------------|---------------------------|--------|\n"
    for name, git_url in sorted_plugins:
        table += f"| {name} | | | {git_url} |\n"
    return table

def main(file_path):
    """
    Main function to read the XML file, process each plugin, and generate the markdown table.
    
    :param file_path: Path to the XML file containing plugin information
    """
    if not os.path.exists("all-plugins"):
        os.makedirs("all-plugins")
        print("Created ./all-plugins directory")

    tree = ET.parse(file_path)
    root = tree.getroot()

    processed_plugins = set()
    plugins_info = []

    for plugin in root.findall('plugin'):
        name = plugin.find('name').text.strip() if plugin.find('name') is not None else ""
        source = plugin.find('source').text.strip() if plugin.find('source') is not None else ""
        
        if name and source and name not in processed_plugins:
            result = process(name, source)
            if result:
                plugins_info.append(result)
            processed_plugins.add(name)
        elif not name or not source:
            print(f"Skipping plugin with missing information: {name or 'Unknown'}")

    # Generate and write the markdown table
    markdown_table = generate_markdown_table(plugins_info)
    with open("plugins.md", "w") as md_file:
        md_file.write(markdown_table)
    
    print("Generated plugins.md with the sorted markdown table.")

if __name__ == "__main__":
    parser = argparse.ArgumentParser(description="Clone OpenCPN plugins from an XML file and generate a sorted markdown table.")
    parser.add_argument("file_path", nargs="?", default="plugins.xml", 
                        help="Path to the XML file containing plugin information (default: plugins.xml)")
    args = parser.parse_args()

    main(args.file_path)

```